app/src/main/java/com/example/a17255/a8project/GamePlay.java

```diff
@@ -92,26 +160,173 @@ public MyCanvas(Context context) {
         protected void onDraw(Canvas canvas) {
             super.onDraw(canvas);
 
+//            draw five obstacles and one target
+            drawObstacle1(canvas, obstacleX1, obstacleY1, obstaclePaint1);
+            drawObstacle1(canvas, obstacleX2, obstacleY2, obstaclePaint2);
+            drawObstacle2(canvas, obstacleX3, obstacleY3, obstaclePaint3);
+            drawObstacle2(canvas, obstacleX4, obstacleY4, obstaclePaint4);
+            drawObstacle2(canvas, obstacleX5, obstacleY5, obstaclePaint5);
+            drawTarget(canvas, targetX, targetY, targetPaint);
+
             if (currentSpeedX == 0 && currentSpeedY == 0) {
                 drawBall(canvas);
             } else {
-
-                mCurrentX = mCurrentX + currentSpeedX / 1000 * DURATION_PER_FRAME_IN_MS;
-                mCurrentY = mCurrentY + currentSpeedY / 1000 * DURATION_PER_FRAME_IN_MS;
+//                acceleromater affect the speed
+                if (mCurrentX > 0 && mCurrentX < canvas.getWidth() && mCurrentY > 0 && mCurrentY < canvas.getHeight()){
+                    currentSpeedX = currentSpeedX - accelerometerX*30;
+                    currentSpeedY = currentSpeedY + accelerometerY*30;
+                }
+
+
+
+//                detect screen bound and change direction when reach screen bound
+                if (mCurrentX >= canvas.getWidth()) {
+                    score += 10;
+                    currentSpeedX = -currentSpeedX;
+                }
+                if (mCurrentX <= 0) {
+                    score += 10;
+                    currentSpeedX = -currentSpeedX;
+                }
+                if (mCurrentY >= canvas.getHeight()) {
+                    score += 10;
+                    currentSpeedY = -currentSpeedY;
+                }
+                if (mCurrentY <= 0) {
+                    score += 10;
+                    currentSpeedY = -currentSpeedY;
+                }
+
+//                detect the collision between the ball and the obstacle light gray
+                if (Math.pow(mCurrentX - obstacleX1, 2) + Math.pow(mCurrentY - obstacleY1, 2) <=
```

Comment:
It is better to create a funtion like: `float calculateDistance(x1,y1,x2,y2)` to make the code easier to read and avoid duplication